src/Plugins/LockingMiddleware.php

```diff
@@ -68,4 +68,12 @@ protected function executeQueuedJobs()
 
         return array_shift($returnValues);
     }
+    
+    private function unlockOnException($e): void
+    {
+        $this->isExecuting = false;
+        $this->queue = [];
+        throw $e;
```

Comment:
Detail, but: re-throwing should probably be done in the catch clauses directly in order to avoid cluttering the stack trace with a useless item.